Undyne labels, equipable restrictions, room restrictions, etc

Jacky720 opened the pull request:
- Add an "Allowed Locations" dropdown - restricts room selection to non-Dogchecked rooms or SAVE points only. Also, complete room list (for v1.08).
- Likewise with weapons/armors, in checkbox form. Fixes #4 for these options.
- Cowboy Hat, Temmie Armor now correctly affect weapon AT. (#59)
- Correctly label the Undyne flags: General and Undying.
- `insert_inv_lists` and `insert_cell_lists` are only used once each, in the context of similar assignments. Cut the middleman.
- `updateSelection` doesn't need to access the values and the index separately- merge the parameters for clarity with default values.
- There are several if/else branches to set values to booleans. Cut out the middleman by assigning the expression to the value.
- Iterate over cell and inventory spaces in saving rather than manual assignment.
- Add `row-span-2` class. Just in case.

Jacky720 committed:
Add row span class

Jacky720 committed:
Fix Undyne labels

Jacky720 committed:
Fix Undyne states better

Ah, that's the spirit.

Jacky720 committed:
Add non-equipables checkbox

Not functional yet, gonna take another look at the JS.

Jacky720 committed:
Merge pull request #1 from Jacky720/patch-1

Update main.js

Jacky720 committed:
Add equipable arrays, remove insert function

Jacky720 committed:
Make equipable arrays objects; simplify iteration

Initialize i to 1 near line 705 to avoid inline incrementing; iterate over the elements near line 763 rather than spell it out.

Jacky720 committed:
Merge insert_inv_lists with insert_cell_lists

Jacky720 committed:
Simplify checkbox saving; Implement armor ATK

Jacky720 committed:
add comma

Apologies.

Jacky720 committed:
Correct armorAt setting on armor change

Jacky720 committed:
Restrict equipable selection

I'll need to do some testing after this, but it should now handle #4 on the main branch.

Jacky720 committed:
Attempt bugfixing; simplify boolean setting

The code has no right to be running `updateSaveDataForm()` on start, but it's the only thing I can think of to explain why it's still using the items array. Also, we don't need ifs for setting booleans like that.

Jacky720 committed:
Remove single-use insert_inv_lists function

As it will apparently never be called outside of start() and even then would be better-suited along with the other field initializations.

Jacky720 committed:
Add "allowed locations" dropdown

main.js should give this the functionality of choosing between SAVE points only, all rooms, or Dogchecked rooms excluded.

Jacky720 committed:
Fix label width

Why would the file0 data have a fourth column? It's practically the same info.

Jacky720 committed:
Use three columns for allowed locations

That looks better still. Not best... but better.

Jacky720 committed:
Add margin to new div

And move some other margins around for cleanliness.

Jacky720 committed:
Progress on allowed rooms selection

Still non-functional, but I should test the array initialization before I continue. It's also probably bad practice, but it's better than duplicating the array or permanently modifying the original in code.

Jacky720 committed:
Initialize rooms[0] and rooms[1]

Should fix current failure.

Jacky720 committed:
Add allowedLocations functionality

Jacky720 committed:
Remove redundant updateSelection parameter

This cleans up the onchange events a little, since they don't use the full values list. Also default to the SAVE points-only room list.

Jacky720 committed:
Tabs to spaces

Is it really that obvious I switched to an external editor? Anyway, if these tabs don't want to play nice and be four spaces long let's just be rid of them.

Jacky720 committed:
More checkbox simplification; convert to number in saving

KockaAdmiralac commented on main.js:
```diff
@@ -947,20 +1038,53 @@ function start() {
+        allowedLocations2.value = allowedLocations1.value;
+        updateSelection("ini-location", document.getElementById("ini-location").value, rooms[allowedLocations1.value]);
+        updateSelection("sav-location", document.getElementById("sav-location").value, rooms[allowedLocations1.value]);
+    }, false);
```
> If not specified, `useCapture` defaults to `false`.

[MDN](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener). We might not need these parameters passed around the code.

KockaAdmiralac commented on main.js:
```diff
@@ -947,20 +1038,53 @@ function start() {
+    
     var weaponSelect = document.getElementById("sav-weapon");
+    var armorSelect = document.getElementById("sav-armor");
     weaponSelect.onchange = function() {
```
Could this be `weaponSelect.addEventListener("change", function() {...});` like in the rest of the code?

KockaAdmiralac commented on main.js:
```diff
@@ -105,252 +135,361 @@ var ArmorDfs = {
+}
+// rooms[0] is SAVE points only, rooms[1] is non-Dogchecked only, rooms[2] contains all rooms.
+
+var roomSelectOptions = [
+    "SAVE points only",
+    "Accessible rooms only",
+    "All rooms"
+];
```
I think accessible rooms should remain the default option so the new option doesn't confuse previous users.

KockaAdmiralac commented on index.html:
```diff
@@ -129,6 +129,11 @@ <h1>Persistent Data (undertale.ini)</h1>
               id="ini-location"
               class="w-full"
             ></select>
+            <label>Allowed Locations</label>
```
What about "Filter locations to:"? "Allowed Locations" sounds like it might change something in the file regarding what rooms can be accessed, when it's really just an interface change.

KockaAdmiralac commented on index.html:
```diff
@@ -93,8 +93,8 @@ <h1>Persistent Data (undertale.ini)</h1>
-          <div class="grid-cols-3">
-            <label class="mt-1.5">Name</label>
+          <div class="grid-cols-3 mt-1.5">
+            <label>Name</label>
```
If you want to tackle it here, these labels are supposed to have a `for` attribute pointing to the `<input>` element they represent, so clicking the label delegates that event to the input element. For example, here it would be `for="ini-name"`.

Jacky720 committed:
Consistency; rm redundant parameter; default rooms

With suggestions to #72:
- **Consistency:** camelCase "armorDfs"; unnecessary variables, just call getElementById directly; use addEventListener.
- **Redundant parameter:** `false` is the default for that parameter.
- **Location options:** Allow all non-Dogchecked rooms by default.

Jacky720 committed:
Add for tags, change room filter label

Also, correct Goner Kid fun value and Mettaton state title.

Jacky720 committed:
Standard phrasing

(oops)
